### ISSUE TITLE
TypeError: Cannot assign to read only property 'viewBox' of object '#<SVGSVGElement>'

### ISSUE THREAD

**Issue by brainkim** [bikeshaving/crank#117]
https://github.com/bikeshaving/crank/issues/102#issuecomment-632252868
https://codesandbox.io/s/crank-svg-read-only-attribute-issue-9psct?file=/src/index.js

**Comment by jonrsawyer** [comment]
I may have jumped to a conclusion regarding spread attributes. I was assuming that you probably do something like this somewhere:
```
...
newSvgElement.setAttributes({...oldAttributes});
...
```
That is, use the spread operator to transfer old attributes to the new element. If this resulted in an assignment to a read-only attribute, that would trigger an error such as I'm seeing.

**Comment by brainkim** [comment]
Yeah without a reproduction I can bet the error is likely on this line: https://github.com/bikeshaving/crank/blob/master/src/dom.ts?ts=2#L66

What I don’t understand is why `name in el` is returning true for viewBox? This might be browser-specific. I think the fix might be to change the logic to use setAttribute for all svg nodes, but it would be nice to get a deeper understanding of what made this error hppen in the first place.

**Comment by jonrsawyer** [comment]
My first codesandbox - hope I'm sharing it right:

https://codesandbox.io/s/crank-svg-read-only-attribute-issue-9psct?file=/src/index.js

**Comment by brainkim** [comment]
Thank you! I’m able to reproduce it in safari too so something is definitely up. I am currently knee deep in a refactor of the core stuff trying to make Crank not such a memory hog but I will try to triage this bug by midday (2PM EST) tomorrow (Friday). If in the meantime you want to explore it yourself and get that contributor badge next to your name, lemme know! The codebase still isn’t excessively complicated and I would be glad to help you figure out how to get the codebase up and running here or via email/discord (brain#4221). I haven’t written a CONTRIBUTING.md yet so there’s probably some things that aren’t obvious. No pressure though.

**Comment by jonrsawyer** [comment]
I've updated my codesandbox to demonstrate the following: assigning to an SVG attribute is disallowed, apparently:
```
circle.cx = 13; // fails with the read-only error
```
You must use `setAttribute()`:
```
circle.setAttribute("cx", 13);
```

**Comment by brainkim** [comment]
Right so it looks like if we’re in SVG mode we just have to use setAttribute exclusively. Doesn’t explain why viewBox was defined on an SVGSVGElement but the fix should be straightforward. I’ll  get it done by 10PM today.

**Comment by brainkim** [comment]
https://codesandbox.io/s/crank-svg-test-117-b1l2y

Fixed in 0.1.5
